### DISCUSSION TITLE
a question about declaration

### DISCUSSION THREAD

**Discussion by zhefenfei** [zeux/pugixml#509] [General]
when read a file , I wanna save it ,  invoke  ->  doc.save(../test.xml). Then ,the declaration is <? version = "1.0">,original declaration is <? version = "1.0" encoding = "utf-8">,just like the encoding has been lost when read file...

**Comment by zeux** [comment]
By default, declarations are not processed. You need to use `parse_default | parse_declaration` flag during parsing, which will preserve the declaration node along with all its attributes including encoding.
